pypeit/edgetrace.py

```diff
@@ -4992,6 +4994,27 @@ def match_order(self):
         # Find the smallest offset for each order
         slit_indx = np.ma.MaskedArray(np.ma.argmin(np.absolute(sep), axis=1))
 
+        # Cut down, as needed
+        if self.spectrograph.order_spat_pos.size < slit_cen.size:
+            msgs.warn("We detected more orders than calibrated. Trimming the extras")
+            # Mask
+            bad_slits = np.ones(slit_cen.size, dtype=bool)
+            bad_slits[slit_indx] = False
+            bad_indx = np.where(bad_slits)[0]
+            # Flag them
+            _indx = np.append(2*bad_indx, 2*bad_indx+1)
+            self.edge_msk[:,_indx] = self.bitmask.turn_on(
+                self.edge_msk[:,_indx], 'ORDERMISMATCH')
+
+            # Redo the above calculations                                                            
+            slit_cen = slit_cen[np.logical_not(bad_slits)]
+            sep = self.spectrograph.order_spat_pos[:,None] - slit_cen[None,:] - offset
+            slit_indx = np.ma.MaskedArray(np.ma.argmin(np.absolute(sep), axis=1))
+        # TODO -- Should I keep this on, or might we succeed (possible)?
```

Comment:
It can't hurt keeping this in, right?

Reply:
I think the code below might handle it, so I'm going to keep it off for now.